src/idl_parser.cpp

```diff
@@ -897,6 +897,29 @@ CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
     EXPECT(',');
   }
 
+  // Check if all required fields are parsed.
+  for (auto field_it = struct_def.fields.vec.begin();
+            field_it != struct_def.fields.vec.end();
+            ++field_it) {
+    auto required_field = *field_it;
+    if (!required_field->required) {
+      continue;
+    }
+    bool found = false;
+    for (auto pf_it = field_stack_.begin();
```

Comment:
Could you help me understand how this can happen?
I tried schema
`namespace sample;

table NestedObject1 {
	count: int;
	nestedObject: NestedObject (required);
}

table NestedObject {
	count: int;
}

table ParentObject {
	nestedObject: NestedObject (required);
	nestedObject1: NestedObject1 (required);
}

root_type ParentObject;`
with the following json, but it detected that nestedObject was missing eventhough the parent object contains a similar field.
`{
	nestedObject: {
		count : 5
	},
	nestedObject1: {
	}	
}`